Competition2017/src/competition/subsystems/vision/VisionSubsystem.java

```diff
@@ -91,21 +90,32 @@ private void handleCommPacket(OffboardCommPacket packet) {
                 newTarget.isTracked = jsonTarget.getBoolean("isTracked");
 
                 newTarget.distance = jsonTarget.getDouble("distance");
-                
+
                 return newTarget;
             });
         }
     }
-    
-    private <T> void loadJsonArray(List<T> outList, JSONObject payload, String jsonKey, Function<JSONObject, T> objectParser) {
+
+    private <T> void loadJsonArray(List<T> outList, JSONObject payload, String jsonKey,
+            Function<JSONObject, T> objectParser) {
         outList.clear();
-        if(payload.has(jsonKey)) {
+        if (payload.has(jsonKey)) {
             JSONArray jsonTargets = payload.getJSONArray(jsonKey);
-            for(Object targetObj : jsonTargets) {
-                if(JSONObject.class.isInstance(targetObj)) {
-                    outList.add(objectParser.apply((JSONObject)targetObj));
+            for (Object targetObj : jsonTargets) {
+                if (JSONObject.class.isInstance(targetObj)) {
+                    outList.add(objectParser.apply((JSONObject) targetObj));
                 }
             }
         }
     }
+
+    @Override
+    public void updatePeriodicData() {
+        double currentTime = Timer.getFPGATimestamp();
+        if (currentTime - lastTimePacketRecieved >= 5 || packet == null) {
```

Comment:
This current logic will continually log "disconnected" until packet isn't null; we only want to do it once. Add a Boolean for `lastConnectionState` or similar and make it `false` to start; when a packet comes in, make it true, and when there isn't anything received for over the threshold make it false. Whenever it changes value, log appropriately (for both "connected" and "disconnected").